src/inference/elbo.ad.js

```diff
@@ -208,6 +209,54 @@ module.exports = function(env) {
       return k(s);
     },
 
+    mapDataFetch: function(data, batchSize, address) {
+
+      var ix;
+      if (batchSize === data.length) {
+        // Use all the data, in order.
+        ix = [];
+      } else {
+        ix = _.times(batchSize, function() {
+          return Math.floor(util.random() * data.length);
```

Comment:
Would it be better to sample these indices without replacement? (See e.g. [here](http://arxiv.org/pdf/1202.4184v1.pdf).)